Source/Tests/ELFCodeLoader2.h

```diff
@@ -734,19 +756,40 @@ class ELFCodeLoader2 final : public FEXCore::CodeLoader {
     VDSOBase = Base;
   }
 
+  void CalculateHWCaps(FEXCore::Context::Context *ctx) {
+    // HWCAP is just CPUID function 0x1, the EDX result
+    auto res_1 = FEXCore::Context::RunCPUIDFunction(ctx, 1, 0);
+    HWCap = res_1.edx;
+
+    // HWCAP2 is as follows:
+    // Bits:
+    // 0 - MONITOR/MWAIT available in CPL3
+    // 1 - FSGSBASE instructions available in CPL3
+    HWCap2 = 0;
+  }
+
   constexpr static uint64_t BRK_SIZE = 8 * 1024 * 1024;
   constexpr static uint64_t STACK_SIZE = 8 * 1024 * 1024;
 
   std::vector<std::string> Args;
   std::vector<std::string> EnvironmentVariables;
   std::vector<char const*> LoaderArgs;
 
-  std::vector<auxv_t> AuxVariables;
+  std::list<auxv_t> AuxVariables;
```

Comment:
Why change this to `std::list`?

Reply:
When storing references from std::vector emplace_back, when it resizes the internal container the reference is invalidated. Making it impossible to cleanly modify AT_RANDOM and AT_PLATFORM auxv_t values.
So switched it to a list so they don't get invalidated.

Reply:
Ah, that makes sense 👍